src/main/java/org/wso2/carbon/connector/operations/BatchExecute.java

```diff
@@ -0,0 +1,153 @@
+import org.wso2.carbon.connector.utils.Error;
+import org.wso2.carbon.connector.utils.SnowflakeUtils;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+/**
+ * Implements batch execute operation.
+ */
+public class BatchExecute extends AbstractConnector {
+
+    private static final String OPERATION_NAME = "batchExecute";
+    private static final String ERROR_MESSAGE = "Error occurred while performing snowflake:batchExecute operation.";
+
+    @Override
+    public void connect(MessageContext messageContext) {
+        ConnectionHandler handler = ConnectionHandler.getConnectionHandler();
+        String connectionName = null;
+        SnowflakeConnection snowflakeConnection = null;
+
+        try {
+            connectionName = SnowflakeUtils.getConnectionName(messageContext);
+            snowflakeConnection = (SnowflakeConnection) handler.getConnection(Constants.CONNECTOR_NAME, connectionName);
+            SnowflakesOperationResult result = batchExecuteQuery(messageContext, snowflakeConnection);
+            SnowflakeUtils.setResultAsPayload(messageContext, result);
+        } catch (InvalidConfigurationException e) {
+            handleError(messageContext, e, Error.INVALID_CONFIGURATION, ERROR_MESSAGE);
+        } catch (SnowflakeOperationException e) {
+            handleError(messageContext, e, Error.OPERATION_ERROR, ERROR_MESSAGE);
+        } catch (ConnectException e) {
+            handleError(messageContext, e, Error.CONNECTION_ERROR, ERROR_MESSAGE);
+        } finally {
+            if (snowflakeConnection != null) {
+                handler.returnConnection(Constants.CONNECTOR_NAME, connectionName, snowflakeConnection);
+            }
+        }
+    }
+
+    /**
+     * Executes the query.
+     *
+     * @param messageContext      Message Context.
+     * @param snowflakeConnection Snowflake Connection.
+     * @return SnowflakesOperationResult.
+     * @throws InvalidConfigurationException Invalid Configuration Exception.
+     * @throws SnowflakeOperationException  Snowflake Operation Exception.
+     */
+    private SnowflakesOperationResult batchExecuteQuery(MessageContext messageContext, SnowflakeConnection snowflakeConnection)
+            throws InvalidConfigurationException, SnowflakeOperationException {
+        SnowflakesOperationResult snowflakesOperationResult =
+                new SnowflakesOperationResult(OPERATION_NAME, false);
+        String query = (String) getParameter(messageContext, Constants.EXECUTE_QUERY);
+        String payload = (String) getParameter(messageContext, Constants.PAYLOAD);
+
+        if (StringUtils.isEmpty(query)) {
+            throw new InvalidConfigurationException("Execute Query is not provided.");
+        }
+
+        if (StringUtils.isEmpty(payload)) {
+            throw new InvalidConfigurationException("Empty Payload is provided.");
+        }
+
+        JsonArray executeArray = JsonParser.parseString(payload).getAsJsonArray();
```

Comment:
shall we check isJsonArray first ? else it will throw an error

Reply:
Fixed.